crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -284,49 +289,77 @@ impl SolverTimeout {
     }
 }
 
-/// The solution score. This is often referred to as the "objective value".
-#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
-pub struct Score(pub eth::U256);
+/// Carries information how the score should be calculated.
+#[derive(Debug, Clone)]
+pub enum Score {
+    Solver(eth::U256),
+    Discount(eth::U256),
+    RiskAdjusted {
+        success_probability: SuccessProbability,
+        gas_amount: Option<eth::U256>,
+    },
+}
 
-impl From<Score> for eth::U256 {
-    fn from(value: Score) -> Self {
-        value.0
-    }
+/// Represents the probability that a solution will be successfully settled.
+#[derive(Debug, Clone)]
+pub enum SuccessProbability {
+    /// Probability known and is equal to the given value.
+    Value(f64),
```

Comment:
Is it correct that a `SuccessProbability::Value(x)` can be modeled as `SuccessProbability::Params { intercept: x, ..default }`?

Reply:
I don't think so (at least based on the formula for success_probability in the RiskCalculator)